test/server_search.cpp

```diff
@@ -935,6 +953,36 @@ struct TestData
 TEST(ServerSearchTest, searchResults)
 {
   const TestData testData[] = {
+    {
+      /* query */          "pattern=&books.id=" RAYCHARLESZIMID,
+      /* start */            -1,
+      /* resultsPerPage */   0,
+      /* totalResultCount */ 0,
+      /* firstResultIndex */ 0,
+      /* results */          {},
+      /* pagination */       {}
+    },
+
+    {
+      /* query */           "pattern=%20&books.id=" RAYCHARLESZIMID,
+      /* start */             -1,
+      /* resultsPerPage */    0,
+      /* totalResultCount */ 0,
+      /* firstResultIndex */ 0,
+      /* results */          {},
+      /* pagination */       {}
+    },
+
+    {
+      /* query */           "pattern=%20yellow%20&books.id=" RAYCHARLESZIMID,
+      /* start */             -1,
+      /* resultsPerPage */    0,
+      /* totalResultCount */ 2,
+      /* firstResultIndex */ 0,
+      /* results */          YELLOW_SEARCH_RESULTS,
+      /* pagination */       {}
+    },
+
```

Comment:
This change doesn't agree with the summary of the commit message. What does it have to do with data deduplication in existing unit-tests?

Would the tests pass if the code is compiled with the source tree checked out at this commit?

Reply:
This change should be preserved. It just has to belong to the last commit.